core/include/webview/detail/backends/win32_edge.hh

```diff
@@ -681,10 +681,8 @@ protected:
     auto res =
         m_webview->AddScriptToExecuteOnDocumentCreated(wjs.c_str(), &handler);
     if (SUCCEEDED(res)) {
-      // Sadly we need to pump the even loop in order to get the script ID.
-      while (!done) {
-        deplete_run_loop_event_queue();
-      }
+      // Sadly we need to pump the event loop in order to get the script ID.
+      run_event_loop_while([&] { return !done; });
```

Comment:
In terms of #1249, this event depletion will be executed when a user calls `bind`, and if this is done before `set_size`, it will cause the problematic visual glitch. We however still need to handle the `dispatch_size_default` "future" by doing something like:
```c++
// We want to guard against executing the default `set_size` prematurely
default_size_guard(true);
run_event_loop_while([&] { return !done; });
// The user's `set_size` may have been executed from the depleted event queue,
// so guard against putting the default `set_size` back onto the queue.
if (!m_is_window_shown) {
  default_size_guard(false);
  dispatch_size_default(m_owns_window);
}
```

Reply:
Also, see https://github.com/webview/webview/pull/1249#discussion_r2009174449 for the init script scenario.